tests/test_writer_text.py

```diff
@@ -17,43 +17,24 @@
 from __future__ import division
 from __future__ import print_function
 
+from functools import partial
+
 import six
 
 from base64 import b64encode
-from datetime import datetime, timedelta
-from decimal import Decimal
+from datetime import timedelta
 from io import BytesIO
 from itertools import chain
-from pytest import raises
-
-from tests import is_exception, noop_manager, parametrize
-
-from amazon.ion.core import IonType, IonEvent, IonEventType, OffsetTZInfo
-from amazon.ion.symbols import SymbolToken
-from amazon.ion.util import record
-from amazon.ion.writer import blocking_writer, WriteEventType
-from amazon.ion.writer_text import raw_writer
-
-
-_D = Decimal
-_DT = datetime
-
-_E = IonEvent
-_IT = IonType
-_ET = IonEventType
 
 
-class _P(record('desc', 'events', 'expected')):
-    def __str__(self):
-        return self.desc
+from amazon.ion.core import OffsetTZInfo
 
+from tests import parametrize
 
-def _scalar_p(ion_type, value, expected):
-    return _P(
-        desc='SCALAR %s - %s' % (ion_type.name, expected),
-        events=(_E(_ET.SCALAR, ion_type, value),),
-        expected=expected,
-    )
+from amazon.ion.symbols import SymbolToken
+from amazon.ion.writer import blocking_writer
+from amazon.ion.writer_text import raw_writer
+from tests.writer_util import assert_writer_events, _E, _ET, _DT, _IT, _D, _P, _generate_scalars, _generate_containers
```

Comment:
Removed from the utility module and moved back to the tests where they are used.